### ISSUE TITLE
Invalid Region Constant in Aws::Region (AP_NORTHEAST_4) & Missing Region Constant (AP_SOUTHEAST_4)

### ISSUE THREAD

**Issue by James-02** [aws/aws-sdk-cpp#3057]
### Describe the bug

Within `Aws::Region` there is a constant `AP_NORTHEAST_4`, which translates to `ap-northeast-4`, however this is not an actual region available in AWS: https://docs.aws.amazon.com/general/latest/gr/s3.html

### Expected Behavior

The `AP_NORTHEAST_4` constant should actually be `AP_SOUTHEAST_4`, mapping to `ap-southeast-4`

### Current Behavior

N/A

### Reproduction Steps

```C++
#include <aws/core/Aws.h>

std::cout << Aws::Region::AP_NORTHEAST_4 << std::endl;

```

### Possible Solution

Rename the invalid constant to `AP_SOUTHEAST_4` and it's string value to `ap-southeast-4`

### AWS CPP SDK version used

1.11.352

### Compiler and Version used

gcc (GCC) 11.4.1 20231218 (Red Hat 11.4.1-3)

### Operating System and version

Rocky Linux 9.3

**Comment by sbiscigl** [comment]
Yep thats what we get for hand editing it, created a [PR](https://github.com/aws/aws-sdk-cpp/pull/3059) to code generate the enum based on the regions defined in `parititons.json`. Will merge on monday pending any action items on the PR. Thanks for the issue!
